pkg/controller/jobframework/integrationmanager.go

```diff
@@ -265,7 +267,13 @@ func GetIntegration(name string) (IntegrationCallbacks, bool) {
 func GetIntegrationByGVK(gvk schema.GroupVersionKind) (IntegrationCallbacks, bool) {
 	for _, name := range manager.getList() {
 		integration, ok := GetIntegration(name)
-		if ok && gvk == integration.NewJob().GVK() {
+		var g GVKer
```

Comment:
Can you explain why this is needed? Also, this check could be done only when `ok` - so you could do `if ok && matchingGVK(integration, gvk)`

Reply:
>  Also, this check could be done only when ok - so you could do if ok && matchingGVK(integration, gvk)

The check on ok passes, but the ingetration.NewJob() is nil for deployment_controller. And like @vladikkuzn said setting NewJob to a non-nil value means implementing GenericJob interface for a deployment_controller which didn't make sense. The idea was that GenericJob is a big interface, decomposing it into smaller ones will help with other no-op controllers like deployment_controller in the future.

Reply:
Thanks for explaining, sgtm. In the second part of the comment I meant just a small refactor:

```golang
func matchingGVK(integration, gvk) bool {
  if integration.NewJob != nil {
      return gvk == integration.NewJob().GVK()
   } else {
    return gvk == integration.GVKer().GVK()
  }
}
```
then use it `if ok && gvk == matchingGVK(integration, gvk) {`, to avoid calling NewJob if `!ok`. Would this work?